state-chain/pallets/cf-pools/src/lib.rs

```diff
@@ -1006,16 +1007,16 @@ pub mod pallet {
 }
 
 impl<T: Config> SwappingApi for Pallet<T> {
-	fn take_network_fee(input: AssetAmount) -> AssetAmount {
+	fn take_network_fee(input: AssetAmount) -> (AssetAmount, AssetAmount) {
```

Comment:
imo, a little struct would be better for the return type, so we get explicit names like `network_fee` and `remaining_amount` or something => less likely to make mistakes given they're the same type. - and also we can remove the ugly `.0` usage